mailbox/store/src/test/java/org/apache/james/mailbox/store/PreDeletionHooksTest.java

```diff
@@ -171,12 +211,20 @@ void runHooksShouldExecuteHooksSequentially() {
     }
 
     @Test
-    void runHooksShouldPublishTimerMetrics() {
-        when(hook1.notifyDelete(any())).thenReturn(Mono.empty());
-        when(hook2.notifyDelete(any())).thenReturn(Mono.empty());
+    void runHooksShouldPublishTimerMetrics() throws Exception {
+        long sleepDurationInMs = Duration.ofSeconds(1).toMillis();
+        Mono monoAnswer = Mono.fromCallable(() -> {
+            Thread.sleep(sleepDurationInMs);
+            return Mono.empty();
+        });
+
+        when(hook1.notifyDelete(any())).thenReturn(monoAnswer);
+        when(hook2.notifyDelete(any())).thenReturn(monoAnswer);
```

Comment:
returning the same mono from two different hooks seem weird? could you return two different mono from two hook calls?

Reply:
I think no need, I've seen the assertion, and it's good